# NumPy Business Case: Retail Sales Performance Analysis

## Business Scenario

A retail company wants to analyze its sales performance during a 30-day period.

The objective is to understand:

- Total revenue
- Average daily sales
- Best and worst performing days
- Sales variability
- Days with above-average performance

NumPy will be used to generate and analyze the dataset.

## Dataset Generation

In [37]:
import numpy as np

np.random.seed(42)

days = np.arange(1,31)

daily_sales = np.random.randint(
    1000,
    5000,
    size=30
)

customers = np.random.randint(
    50,
    200,
    size=30
)

average_ticket = daily_sales / customers

In [36]:
daily_sales[:10]

array([4174, 4507, 1860, 2294, 2130, 2095, 4772, 4092, 2638, 3169])

The generated dataset represents:

- 30 business days
- Daily revenue
- Number of customers
- Average customer spending

## Data Exploration

### First values

In [35]:
print(daily_sales[:10])

[4174 4507 1860 2294 2130 2095 4772 4092 2638 3169]


### Dataset dimensions

In [34]:
print("Number of days:", daily_sales.size)
print("Dimensions:", daily_sales.ndim)

Number of days: 30
Dimensions: 1


## Descriptive Statistics

### Total revenue

In [33]:
total_revenue = np.sum(daily_sales)

print(total_revenue)

91462


### Average daily sales

In [32]:
average_sales = np.mean(daily_sales)

print(average_sales)

3048.733333333333


### Median

In [31]:
median_sales = np.median(daily_sales)

print(median_sales)

2910.0


### Minimum and maximum

In [30]:
minimum_sales = np.min(daily_sales)

maximum_sales = np.max(daily_sales)


print(minimum_sales)
print(maximum_sales)

1130
4772


## Standard deviation

In [29]:
sales_std = np.std(daily_sales)

print(sales_std)

1110.5114117178425


Standard deviation helps us understand sales variability.

A high value means sales fluctuate significantly between days.

## Business KPIs

### Best sales day

In [28]:
best_day = np.argmax(daily_sales)


print(
    "Best day:",
    days[best_day]
)

print(
    "Revenue:",
    daily_sales[best_day]
)

Best day: 7
Revenue: 4772


### Worst sales day

In [27]:
worst_day = np.argmin(daily_sales)


print(
    "Worst day:",
    days[worst_day]
)

print(
    "Revenue:",
    daily_sales[worst_day]
)

Worst day: 20
Revenue: 1130


### Average customers

In [26]:
average_customers = np.mean(customers)

print(average_customers)

113.23333333333333


### Average ticket

In [25]:
average_ticket_value = np.mean(average_ticket)

print(average_ticket_value)

31.10825372939486


## Business Questions

### Which days performed above average?

In [24]:
above_average_days = days[
    daily_sales > average_sales
]


print(above_average_days)

[ 1  2  7  8 10 15 16 17 18 19 22 24 26 27 28]


### What percentage of days exceeded average sales?

In [21]:
percentage = (
    np.sum(daily_sales > average_sales)
    /
    len(daily_sales)
) * 100


print(percentage)

50.0


### What was the revenue difference between the best and worst day?

In [22]:
difference = (
    maximum_sales -
    minimum_sales
)

print(difference)

3642


## Insights and Conclusions

In [38]:
business_summary = {
    "Total revenue": total_revenue,
    "Average daily revenue": average_sales,
    "Best day revenue": maximum_sales,
    "Worst day revenue": minimum_sales,
    "Above average sales days (%)": percentage
}

print(business_summary)

{'Total revenue': 91462, 'Average daily revenue': 3048.733333333333, 'Best day revenue': 4772, 'Worst day revenue': 1130, 'Above average sales days (%)': 50.0}


### Business Insights

After analyzing the simulated retail dataset:

- The company generated a total revenue of **91,462€**.
- The average daily revenue was **3048.73€**.
- The best performing day generated **4772€**.
- Sales variability shows how much daily revenue fluctuates during the period.
- **50% of the days exceeded the average sales level.**


These insights could help business teams identify high-performing periods and support decision-making.

In a real business environment, this analysis would be the first step of an end-to-end analytics workflow, where Python is used for data preparation and analysis, and Power BI is used to build interactive dashboards and communicate insights to stakeholders.